src/framework/audio/internal/worker/audioengine.cpp

```diff
@@ -137,22 +119,24 @@ unsigned int AudioEngine::sampleRate() const
     return m_sampleRate;
 }
 
-std::shared_ptr<IAudioBuffer> AudioEngine::buffer() const
+IAudioBufferPtr AudioEngine::buffer() const
 {
     ONLY_AUDIO_WORKER_THREAD;
     return m_buffer;
 }
 
-std::shared_ptr<IMixer> AudioEngine::mixer() const
+IMixerPtr AudioEngine::mixer() const
 {
-    ONLY_AUDIO_WORKER_THREAD;
+    ONLY_AUDIO_MAIN_OR_WORKER_THREAD;
+
     return m_mixer;
 }
 
-std::shared_ptr<ISequencer> AudioEngine::sequencer() const
+void AudioEngine::setMixer(IMixerPtr mixerPtr)
 {
-    ONLY_AUDIO_WORKER_THREAD;
-    return m_sequencer;
+    ONLY_AUDIO_MAIN_OR_WORKER_THREAD;
+
+    m_mixer = mixerPtr;
```

Comment:
If it can be called from more than one thread, then synchronization is needed.
I would recommend making it so that it is called from only one thread. For example, objects were created and initialized in the thread in which they should be used

Reply:
the only reason of that is the registration of IMixer for IOC, so that it'll be able in other audio classes via INJECT

Reply:
hold on, maybe I was wrong here with ONLY_AUDIO_MAIN...

Reply:
yeah, it could be just ONLY_AUDIO_WORKER_THREAD

I don't actually recall why I've not put it here